File: GDA/src/main/java/programmingtheiot/data/SystemPerformanceData.java
/**
 * This class is part of the Programming the Internet of Things project.
 * 
 * It is provided as a simple shell to guide the student and assist with
 * implementation for the Programming the Internet of Things exercises,
 * and designed to be modified by the student as needed.
 */ 

package programmingtheiot.data;

import programmingtheiot.common.ConfigConst;

import java.io.Serializable;

/**
 * Shell representation of class for student implementation.
 *
 */
public class SystemPerformanceData extends BaseIotData implements Serializable
{
	// static
	
	
	// private var's
	private float CpuUtilization = 0;
	private float DiskUtilization = 0;
	private float MemoryUtilization = 0;
    
	// constructors
	
	public SystemPerformanceData()
	{
		super();
		super.setName(ConfigConst.SYS_PERF_DATA);
	}
	
	
	// public methods
	public void updateData(SystemPerformanceData data)
	{
		// TODO: update local var's
		super.updateData(data);
		this.CpuUtilization = data.CpuUtilization;
		this.DiskUtilization = data.DiskUtilization;
		this.MemoryUtilization = data.MemoryUtilization;
	}
	
	public float getCpuUtilization() {

		return CpuUtilization;
	}
	
	public float getDiskUtilization() {

		return DiskUtilization;
	}
	
	public float getMemoryUtilization() {

		return MemoryUtilization;
	}
	
	public void setCpuUtilization(float val) {
		CpuUtilization = val;
	}
	
	public void setDiskUtilization(float val) {
		DiskUtilization = val;
	}
	
	public void setMemoryUtilization(float val) {
		MemoryUtilization = val;
	}
	
	
	// protected methods
	
	/* (non-Javadoc)
	 * @see programmingtheiot.data.BaseIotData#handleToString()
	 */
	protected String handleToString()
	{
		return null;
	}
	
	/* (non-Javadoc)
	 * @see programmingtheiot.data.BaseIotData#handleUpdateData(programmingtheiot.data.BaseIotData)
	 */
	protected void handleUpdateData(BaseIotData data)
	{
	}
	
}

File: CDA/src/main/python/programmingtheiot/data/SystemPerformanceData.py
#####
# 
# This class is part of the Programming the Internet of Things project.
# 
# It is provided as a simple shell to guide the student and assist with
# implementation for the Programming the Internet of Things exercises,
# and designed to be modified by the student as needed.
#
from src.main.python.programmingtheiot.common import ConfigConst
from src.main.python.programmingtheiot.data.BaseIotData import BaseIotData


class SystemPerformanceData(BaseIotData):
	"""
	Shell representation of class for student implementation.
	
	"""
	DEFAULT_VAL = 0.0

	def __init__(self, d=None):
		"""
		Constructor.

		@param d Defaults to None. The data (dict) to use for setting all parameters.
		It's provided here as a convenience - mostly for testing purposes. The utility
		in DataUtil should be used instead.
		"""
		super(SystemPerformanceData, self).__init__(name=ConfigConst.NOT_SET, d=d)

		if d:
			self.cpuUtil = d['cpuUtil']
			self.diskUtil = d['diskUtil']
			self.memUtil = d['memUtil']
		else:
			self.cpuUtil = self.DEFAULT_VAL
			self.diskUtil = self.DEFAULT_VAL
			self.memUtil = self.DEFAULT_VAL

	def __str__(self):
		"""
		Returns a string representation of this instance.

		@return The string representing this instance.
		"""

		customStr = \
			str('CpuUtilization=' + str(self.getCpuUtilization()) + \
				'DiskUtilization=' + str(self.getDiskUtilization()) +
				'MemoryUtilization=' + str(self.getMemoryUtilization()))

		return customStr
	def getCpuUtilization(self):
		return self.cpuUtil
	
	def getDiskUtilization(self):
		return self.diskUtil
	
	def getMemoryUtilization(self):
		return self.memUtil
	
	def setCpuUtilization(self, cpuUtil):
		self.cpuUtil = cpuUtil
	
	def setDiskUtilization(self, diskUtil):
		self.diskUtil = diskUtil
	
	def setMemoryUtilization(self, memUtil):
		self.memUtil = memUtil
	
	def _handleUpdateData(self, data):
		if data != None:
			self.cpuUtil = data.cpuUtil
			self.diskUtil = data.diskUtil
			self.memUtil = data.memUtil
